fedimint-testing/src/gateway.rs

```diff
@@ -29,56 +37,58 @@ impl GatewayTest {
         GatewayRpcClient::new(self.api.clone(), self.password.clone())
     }
 
+    /// Removes a client from the gateway
+    pub async fn remove_client(&self, fed: &FederationTest) -> Arc<Client> {
+        self.gateway.remove_client(fed.id()).await.unwrap()
+    }
+
     /// Connects to a new federation and stores the info
-    pub async fn connect_fed(&mut self, fed: &FederationTest) {
+    pub async fn connect_fed(&mut self, fed: &FederationTest) -> FederationInfo {
         let connect = fed.connection_code().to_string();
-        let client = self.get_rpc().await;
-        client
-            .connect_federation(ConnectFedPayload { connect })
+        let rpc = self.get_rpc().await;
+        rpc.connect_federation(ConnectFedPayload { connect })
             .await
-            .unwrap();
-    }
-
-    /// Returns the last registration we sent to a fed
-    pub async fn last_registered(&self) -> LightningGateway {
-        let info = self.get_rpc().await.get_info().await;
-        let fed = info.expect("Failed to get_info").federations;
-        fed.last().expect("No feds registered").registration.clone()
+            .unwrap()
     }
 
     pub(crate) async fn new(
         base_port: u16,
         password: String,
-        lightning: impl ILnRpcClient + 'static,
+        lightning: Arc<dyn LightningTest>,
         decoders: ModuleDecoderRegistry,
```

Comment:
Can we get the decoders from the registry now that we're passing that in?

Reply:
Seems like a good idea